codegen/core/src/main/scala/com/lightbend/akkasls/codegen/MessageType.scala

```diff
@@ -20,20 +20,58 @@ import scala.jdk.CollectionConverters._
 
 import com.google.protobuf.Descriptors
 
+trait MessageType {
+  def name: String
+  def packageName: String
+  def deriveName(derive: String => String): MessageType
+
+  def fullName: String
+
+  def typeImport = s"$packageName.$fullName"
+
+  /**
+   * The fully qualified type name as seen in generated Java/Scala code.
+   */
+  def fullyQualifiedName = s"$packageName.$name"
+
+  /**
+   * 'base' name of the file that will contain this MessageType relative to the package root,
+   * i.e. 'com/example/MyService'
+   */
+  def fileBasename: String =
+    packageName.replace('.', '/') + "/" + name
+}
+
+case class PojoMessageType(packageName: String, name: String) extends MessageType {
+
+  override def deriveName(derive: String => String): MessageType =
+    copy(name = derive(name))
+
+  override def fullName: String = name
+}
+
 /**
- * A fully qualified name that can be resolved in any target language
- *
- * @param descriptorObject
- *   code representation of the descriptor that defined this type, if any
+ * A message type representing a simple Java/Scala type
+ */
+object PojoMessageType {
```

Comment:
`ClassMessageType`, `NativeClassMessageType`, `NativeMessageType`?

Reply:
`ClassMessageType` sounds good. Generic enough and unbiased.